src/main.rs

```diff
@@ -196,26 +214,58 @@ fn main() {
         .cloned()
         .collect();
 
-    let mut cargo = Command::new(cargo_bin());
-
-    cargo.envs(std::env::vars_os());
-
-    cargo
-        .arg("test")
-        .args(first_args)
-        .arg("--")
-        .args(second_args);
+    // Build the final cargo test command.
+    let mut cargo = format!(
+        "{} {} {} -- {}",
+        cargo_bin(),
+        custom_cargo_command.as_deref().unwrap_or_else(|| "test"),
+        first_args
+            .iter()
+            .map(|x| x.to_string_lossy())
+            .fold(String::new(), |mut acc, f| {
+                acc.push(' ');
+                acc.push_str(&f);
+                acc
+            }),
+        second_args
+            .iter()
+            .map(|x| x.to_string_lossy())
+            .fold(String::new(), |mut acc, f| {
+                acc.push(' ');
+                acc.push_str(&f);
+                acc
+            }),
+    );
 
     // We do this weird shit here since cargo does not support running multiple
     // tests individually.
     for item in final_items {
-        cargo.args(["--skip", &item]);
+        cargo.push_str(" --skip ");
+        cargo.push_str(&item);
     }
 
-    cargo.arg("--exact");
+    cargo.push_str(" --exact");
+
+    let mut cmd = {
+        #[cfg(target_family = "windows")]
+        {
+            let mut cmd = Command::new("cmd");
+            cmd.arg("/c");
+            cmd
+        }
+
+        // Assume `sh` is available everywhere else
+        // and let it fail if not
+        #[cfg(not(target_family = "windows"))]
+        {
+            let mut cmd = Command::new("sh");
+            cmd.arg("-c");
+            cmd
+        }
+    };
```

Comment:
> cargo qtest -c "watch -- cargo test"

This approach seems wrong. All we need to do is handle the addition of 1 extra argument next to `qtest` (e.g., `cargo qtest watch` or `cargo-qtest watch`) by checking if the argument exists as a cargo plugin or not, and then use that plugin in the final cargo command.

Here is the final command which we invoke:

https://github.com/onur-ozkan/cargo-qtest/blob/bc5efc5cd47781d15ececfcb235f120fd547c45b/src/main.rs#L199

Reply:
I'm not grasping how to make that work, or I'm not understanding what you're proposing. Hopefully I can illustrate where I'm getting stuck:

#### Current state
```shell
# Input command
$ cargo qtest --all-features -- --nocapture

# qtest executes the following:
cargo test --all-features -- --list --format terse --nocapture

# user is prompted and selects test1
> Search and select set of tests you wish to execute:
+ test1
- test2

# qtest executes the following:
cargo test --all-features -- --nocapture --skip test2 --exact

# test(s) are run and everything works. Great!
# ...but I don't want to have to select tests every time
```

#### My understanding of your suggestion
```shell
# Input command (with the addition of a single positional argument, here illustrated using `watch`)
$ cargo qtest watch --all-features -- --nocapture

# qtest executes the following (same as before):
cargo test --all-features -- --list --format terse --nocapture

# user is prompted and selects test1 (same as before)
> Search and select set of tests you wish to execute:
+ test1
- test2

# qtest "remembers" the extra argument to use instead of(?) `test`
cargo watch --all-features -- --nocapture --skip test2 --exact

# Error because those are not valid arguments to `cargo watch`
```

#### My proposal
```shell
# Input command (with the addition of a single argument `-c` with required value, here illustrated using `"watch -- cargo test"`)
$ cargo qtest -c "watch -- cargo test" --all-features -- --nocapture

# qtest executes the following (same as before):
cargo test --all-features -- --list --format terse --nocapture

# user is prompted and selects test1 (same as before)
> Search and select set of tests you wish to execute:
+ test1
- test2

# qtest "remembers" the extra argument to use instead of `test`
cargo watch -- cargo test --all-features -- --nocapture --skip test2 --exact

# test(s) are run and everything works
# execution waits for cargo-watch to rerun the tests on changes the workspace files
```

Reply:
Hmm I see, I didn't know how `watch` works before. Your proposal is much better in this case.